homeassistant/components/travel_time/__init__.py

```diff
@@ -0,0 +1,193 @@
+"""The travel_time integration."""
+from datetime import timedelta
+import logging
+from typing import Dict, Optional, Union
+
+import voluptuous as vol
+
+from homeassistant.const import ATTR_ATTRIBUTION, ATTR_LATITUDE, ATTR_LONGITUDE
+from homeassistant.core import State
+from homeassistant.helpers import location
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.loader import bind_hass
+
+from .const import (
+    ATTR_DESTINATION,
+    ATTR_DESTINATION_NAME,
+    ATTR_DISTANCE,
+    ATTR_DURATION,
+    ATTR_DURATION_IN_TRAFFIC,
+    ATTR_MODE,
+    ATTR_ORIGIN,
+    ATTR_ORIGIN_NAME,
+    ATTR_ROUTE,
+    ATTR_ROUTE_MODE,
+    ATTR_TRAFFIC_MODE,
+    ATTR_UNIT_SYSTEM,
+    DOMAIN,
+    UNIT_OF_MEASUREMENT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+SCAN_INTERVAL = timedelta(minutes=5)
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({}, extra=vol.ALLOW_EXTRA)
```

Comment:
We should specify a standardized schema that platforms have to follow to specify source and target location?

Reply:
1. For that I would like to have https://github.com/home-assistant/home-assistant/pull/30019 merged first.

2. Moreover the current travel_time integrations provide support for at least coordinates. Waze and Google allow names and Google allows for Google Place Ids as well.
All integrations allow to supply entity_ids like device_trackers etc.

    Google and Waze both only have `destination` and `origin` which can hold everything.

    For HERE we introduced separate `origin_latitude`, `origin_entity_id`, and so on. The 
    discussion can be found here: https://github.com/home-assistant/home- 
    assistant/pull/24603#discussion_r311280860
    
    So I am not sure what should be standard given all these options.

3. Given the above I implemented the check that exactly one method is used for `here_travel_time` in the following way: https://github.com/home-assistant/home-assistant/blob/7ae537304e8f8bc548c1f67b997b06fe07bb6da8/homeassistant/components/here_travel_time/travel_time.py#L83-L90 but do not know how to put that in a base schema. All the ways I tried lead to "schema cannot be extended because it is not a `dict`"

Reply:
#30019 can be added later. 

It's up to the travel time integration to provide a way to specify the origin and destination. Platforms just need to provide travel time.

The standard should be to accept one of those options. Like you did with `here_travel_time`.

Reply:
So this means no standard platform schema in `travel_time`. Did I understand you correct?
I am not sure whether your initial comment on this was a question whether we should do so or your opinion that we should.

Reply:
No I think we should have a standard platform schema and standardize how people define origin and destination. If we don't do that, then why standardize anything at all ?

Reply:
I implemented a standard TRAVEL_TIME_SCHEMA.
It allows origin and destination to either be defined by a name or a combination of latitude and longitude.
Moreover each of them can be a template:

```yaml
origin_latitude: 0.0
origin_longitude: {{ state_attr('device_tracker.paulus', 'longitude') }}
destination_name: Berlin
```